cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1100,13 +1101,14 @@ func gatherNetworksFromSpec(ds *datastore.RethinkStore, allocationSpec *machineA
 	// what do we have to prevent:
 	// - user wants to place his machine in a network that does not belong to the project in which the machine is being placed
 	// - user wants a machine with a private network that is not in the partition of the machine
-	// - user wants to define multiple private networks for his machine
-	// - user must define one private network
+	// - user specifies no private network
+	// - user specifies multiple primary private networks (those are not shared)
```

Comment:
Is this correct, can a machine have multiple primary not shared networks, i doubt not

Reply:
Multiple primary networks are not allowed; we should use the same logic for all items of the list

Reply:
This is already the case - nothing to do here

```
        // what do we have to prevent:
	// - user wants to place his machine in a network that does not belong to the project in which the machine is being placed
	// - user wants a machine with a private network that is not in the partition of the machine
	// - user specifies no private network
	// - user specifies multiple primary private networks (those are not shared)
	// - user specifies administrative networks, i.e. underlay or privatesuper networks
	// - user's private network is specified with noauto, which would make the machine have no ip address
```

Reply:
got it thanks